src/us/kbase/workspace/database/provenance/RelationshipType.java

```diff
@@ -2,72 +2,166 @@
+	IS_MANIFESTATION_OF("Crossref", "IsManifestationOf"),
+	IS_MANUSCRIPT_OF("Crossref", "IsManuscriptOf"),
+	IS_PREPRINT_OF("Crossref", "IsPreprintOf"),
+	IS_RELATED_MATERIAL("Crossref", "IsRelatedMaterial"),
+	IS_REPLACED_BY("Crossref", "IsReplacedBy"),
+	IS_REPLY_TO("Crossref", "IsReplyTo"),
+	IS_REVIEW_OF("Crossref", "IsReviewOf"),
+	IS_SAME_AS("Crossref", "IsSameAs"),
+	IS_TRANSLATION_OF("Crossref", "IsTranslationOf"),
+	REPLACES("Crossref", "Replaces");
+
 
-	private final static String SOURCE = "DataCite";
 	private final String identifier;
+	private final String source;
+
+	// From the DataCite / Crossref "related identifier" crosswalk
+	// The following are Crossref relationship type terms that duplicate
+	// DataCite terms. If any of the following terms is entered,
+	// it should be replaced by the DataCite term.
+	private static final List<String> CROSSREF_DUPLICATES = Arrays.asList(
+		"Compiles",
+		"Continues",
+		"Documents",
+		"HasPart",
+		"HasVersion",
+		"IsCompiledBy",
+		"IsContinuedBy",
+		"IsDerivedFrom",
+		"IsDocumentedBy",
+		"IsIdenticalTo",
+		"IsOriginalFormOf",
+		"IsPartOf",
+		"IsReferencedBy",
+		"IsRequiredBy",
+		"IsSupplementedBy",
+		"IsSupplementTo",
+		"IsVariantFormOf",
+		"IsVersionOf",
+		"References",
+		"Requires"
+	);
 
 	// mapping of PIDs and related strings to RelationshipType
 	private static final Map<String, RelationshipType> STRING_TO_REL_TYPE_MAP = new HashMap<>();
 	static {
 		for (final RelationshipType rt : RelationshipType.values()) {
+			// is_original_form_of
 			STRING_TO_REL_TYPE_MAP.put(rt.name().toLowerCase(), rt);
-			STRING_TO_REL_TYPE_MAP.put(rt.name().replace("_", "").toLowerCase(), rt);
-			STRING_TO_REL_TYPE_MAP.put(SOURCE.toLowerCase() + ":" + rt.name().toLowerCase(), rt);
+			// isoriginalformof
+			STRING_TO_REL_TYPE_MAP.put(rt.getIdentifier().toLowerCase(), rt);
+			// datacite:is_original_form_of
+			STRING_TO_REL_TYPE_MAP.put(rt.getSource().toLowerCase() + ":" + rt.name().toLowerCase(), rt);
+			// datacite:isoriginalformof
 			STRING_TO_REL_TYPE_MAP.put(rt.getPid().toLowerCase(), rt);
+			// add in the crossref equivalent (if applicable)
+			if (CROSSREF_DUPLICATES.contains(rt.getIdentifier())) {
+				// e.g. crossref:is_variant_form_of
+				STRING_TO_REL_TYPE_MAP.put("crossref:" + rt.name().toLowerCase(), rt);
+				// crossref:isvariantformof
+				STRING_TO_REL_TYPE_MAP.put("crossref:" + rt.getIdentifier().toLowerCase(), rt);
+			}
 		}
 	}
 
-	private RelationshipType(final String identifier) {
+	private RelationshipType(final String source, final String identifier) {
+		this.source = source;
 		this.identifier = identifier;
 	}
 
+	/**
+	 * Get the source (e.g. DataCite or Crossref) of this relationship type.
+	 *
+	 * @return the source.
+	 */
+	private String getSource() {
+		return source;
+	}
+
+	/**
+	 * Get the identifier for this relationship type.
+	 *
+	 * @return the identifier.
+	 */
+	private String getIdentifier() {
+		return identifier;
+	}
```

Comment:
There's no real need for these methods I think - you can just access via `rt.source` or `rt.identifier` IIRC

Reply:
:+1: